lib/si-layer-cache/src/db/func_run.rs

```diff
@@ -89,6 +91,28 @@ impl FuncRunDb {
                 LIMIT 1
             "#
             ),
+            paginated_workspace_query_with_cursor: format!(
+                r#"
+                SELECT * FROM {DBNAME}
+                WHERE workspace_id = $1 
+                  AND change_set_id = $2
+                  AND (
+                    created_at < (SELECT created_at FROM {DBNAME} WHERE key = $3) OR
+                    (created_at = (SELECT created_at FROM {DBNAME} WHERE key = $3) AND key < $3)
+                )
+                ORDER BY created_at DESC, key DESC
+                LIMIT $4
+                "#
```

Comment:
Yeah, I think that looks about right. I'm wondering if we should consider creating an index related to `created_at` to make the inner queries a bit peppier. @jhelwig does that make sense?

Reply:
I followed up with @britmyerss and @jobelenus about this after a conversation with @jhelwig, thanks all!